Conditional Rendering translation (#51)
reactjs/pt-BR.reactjs.org: content/docs/conditional-rendering.md

---
id: conditional-rendering
title: Renderização condicional
permalink: docs/conditional-rendering.html
prev: handling-events.html
next: lists-and-keys.html
redirect_from:
  - "tips/false-in-jsx.html"
---

Em React, você pode criar componentes distintos que encapsulam o comportamentamento que você precisa. Então, você pode renderizar apenas alguns dos elementos, dependendo do estado da sua aplicação. 

> Renderização condicional em React funciona da mesma forma que condições funcionam em JavaScript. Use operadores de JavaScript  como [`if`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/if...else) ou [operador condicional](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) para criar elementos representando o estado atual, and deixe React atualizar a UI para corresponde-los.

Reviewer: ```suggestion
Renderização condicional em React funciona da mesma forma que condições funcionam em JavaScript. Use operadores de JavaScript  como [`if`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Statements/if...else) ou [operador condicional](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Operators/Operador_Condicional) para criar elementos representando o estado atual, e deixe o React atualizar a UI para corresponde-los.
```

}
```

> Nós vamos criar um componente `Greeting` que mostra um dos outros dois componentes dependendo se o usuário está logado ou não:

Reviewer 2: ```suggestion
Nós vamos criar um componente `Greeting` que mostra um dos outros dois componentes se o usuário estiver logado ou não:
```

);
```

[**Experimente no CodePen**](https://codepen.io/gaearon/pen/ZpVxNq?editors=0011)

> Este exemplo renderiza um "greeting" diferente dependendo do valor da `isLoggedIn` prop.

Reviewer 3: ```suggestion
Este exemplo renderiza um "greeting" diferente dependendo do valor da prop `isLoggedIn`.
```

> ### Variáveis de Elementos {#variáveis-de-elementos}

Reviewer 2: Links internos não precisam ser traduzidos (#42).

> ### Variáveis de Elementos {#variáveis-de-elementos}

Reviewer 2: ```suggestion
### Variáveis de Elementos {#element-variables}
```

> ### Variáveis de Elementos {#variáveis-de-elementos}

Reviewer: It was agreed not to translate internal links.

```suggestion
### Variáveis de Elementos {#element-variables}
```

Author: Thanks for pointing that out. I was confused about translating the links.

}
```

> No exemplo abaixo, nós vamos criar um [componente stateful](/docs/state-and-lifecycle.html#adding-local-state-to-a-class) chamado `LoginControl`.

Reviewer 2: ```suggestion
No exemplo abaixo, nós vamos criar um [componente _stateful_](/docs/state-and-lifecycle.html#adding-local-state-to-a-class) chamado `LoginControl`.
```

);
```

[**Experimente no CodePen**](https://codepen.io/gaearon/pen/QKzAgB?editors=0010)

Declarar uma variável e usar uma declaração condicional `if` é uma ótima maneira de renderizar um componente, mas às vezes você pode querer usar uma sintaxe mais curta. Existem algumas maneiras para utilizar condições inline em JSX, explicadas abaixo.

> ### If em Linha com o Operador Lógico &&  {#inline-if-with-logical--operator}

Reviewer 3: No parágrafo acima, a palavra `inline` não foi traduzida, acho melhor não traduzir aqui também.

```suggestion
### If inline com o Operador Lógico &&  {#inline-if-with-logical--operator}
```

);
```

[**Experimente no CodePen**](https://codepen.io/gaearon/pen/QKzAgB?editors=0010)

Declarar uma variável e usar uma declaração condicional `if` é uma ótima maneira de renderizar um componente, mas 

> às vezes você pode querer usar uma sintaxe mais curta. Existem algumas maneiras para utilizar condições inline em JSX, explicadas abaixo.

Reviewer: I think this ENTER does not exist, may break formatting on site.

Author: Agreed.

> ### Inline If com o Operador Lógico &&  {#inline-if-com-operador--lógico}

Reviewer 2: ```suggestion
### If em Linha com o Operador Lógico &&  {#inline-if-with-logical--operator}
```

> ### Inline If com o Operador Lógico &&  {#inline-if-com-operador--lógico}

Reviewer: ```suggestion
### Inline If com o Operador Lógico &&  {#inline-if-with-logical--operator}
```

);
```

[**Experimente no CodePen**](https://codepen.io/gaearon/pen/ozJddz?editors=0010)

Isto funciona porque em JavaScript, `true && expressão` são sempre avaliadas como `expressão`, e `false && expressão` são sempre avaliadas como `false`.

Portanto, se a condição é `true`, o elemento logo depois do `&&` irá aparecer no resultado. Se o elemento é `false`, React irá ignora-lo.

> ### If-Else em Linha com Operador Condicional {#inline-if-else-with-conditional-operator}

Reviewer 3: Mesmo comentário acima.

```suggestion
### If-Else inline com Operador Condicional {#inline-if-else-with-conditional-operator}
```

Reviewer 4: @WendellAdriel inline tá no glossário?

);
```

[**Experimente no CodePen**](https://codepen.io/gaearon/pen/ozJddz?editors=0010)

> Isto funciona porque em JavaScript, `true && expressão` são sempre avaliadas como `expressão`, é `false && expressão` são sempre avaliadas como `false`.

Reviewer 2: ```suggestion
Isto funciona porque em JavaScript, `true && expressão` são sempre avaliadas como `expressão`, e `false && expressão` são sempre avaliadas como `false`.
```

> Isto funciona porque em JavaScript, `true && expressão` são sempre avaliadas como `expressão`, é `false && expressão` são sempre avaliadas como `false`.

Reviewer: ```suggestion
Isto funciona porque em JavaScript, `true && expressão` são sempre avaliadas como `expressão`, e `false && expressão` são sempre avaliadas como `false`.
```

> Portanto, se a condição é `true`, o elemento logo depois do `&&` irá aparecer no resultado. Se o elemento é `false`, React irá ignora-lo.

Reviewer 2: ```suggestion
Portanto, se a condição é `true`, o elemento logo depois do `&&` irá aparecer no resultado. Se o elemento é `false`, React irá ignorá-lo.
```

> ### Inline If-Else com Operador Condicional {#inline-if-else-com-operador-condicional}

Reviewer 2: ```suggestion
### If-Else em Linha com Operador Condicional {#inline-if-else-with-conditional-operator}
```

> Outro metodo para renderizar elementos inline é utilizar o operador condicional em JavaScript [`condição ? true : false`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Conditional_Operator).

Reviewer 2: ```suggestion
Outro método para renderizar elementos em linha (_inline_) é utilizar o operador condicional em JavaScript [`condição ? true : false`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Operators/Operador_Condicional).
```

> Outro metodo para renderizar elementos inline é utilizar o operador condicional em JavaScript [`condição ? true : false`](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Conditional_Operator).

Reviewer: ```suggestion
Outro método para renderizar elementos inline é utilizar o operador condicional em JavaScript [`condição ? true : false`](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Operators/Operador_Condicional).
```

}
```

> Ele também pode ser usado para expressões mais longas, embora o que está acontecendo seja menos óbvio:

Reviewer 2: ```suggestion
Pode também ser usado para expressões mais longas, embora o que está acontecendo seja menos óbvio:
```

}
```

> Assim como em JavaScript, você decide o estilo apropriado com base no que você e o seu time considera mais legível. Lembre-se  que toda vez que condições se tornam muito complexas, pode ser um bom momento para [extrair um componente](/docs/components-and-props.html#extracting-components).

Reviewer 2: ```suggestion
Assim como em JavaScript, você decide o estilo apropriado com base no que você e a sua equipe considera mais legível. Lembre-se  que toda vez que condições se tornam muito complexas, pode ser um bom momento para [extrair componentes](/docs/components-and-props.html#extracting-components).
```

> ### Evitando que um Componente seja Renderizado {evitando-que-um-componente-seja-renderizado}

Reviewer 2: ```suggestion
### Evitando que um Componente seja Renderizado {#preventing-component-from-rendering}
```

Em casos raros você pode desejar que um componente se esconda ainda que ele tenha sido renderizado por outro componente. Para fazer isso, retorne `null` ao invés do resultado renderizarado.

> No exemplo abaixo, o `<WarningBanner />` é renderizado dependendo do valor da prop chamada `warn`. Se o valor da prop é  `false`, o componente nao é rederizado:

Reviewer: ```suggestion
No exemplo abaixo, o `<WarningBanner />` é renderizado dependendo do valor da prop chamada `warn`. Se o valor da prop é  `false`, o componente não é renderizado:
```

> No exemplo abaixo, o `<WarningBanner />` é renderizado dependendo do valor da prop chamada `warn`. Se o valor da prop é  `false`, o componente nao é rederizado:

Reviewer 2: ```suggestion
No exemplo abaixo, o `<WarningBanner />` é renderizado dependendo do valor da prop chamada `warn`. Se o valor da prop é  `false`, o componente não é renderizado:
```
